src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -49,8 +50,24 @@ def create_configstore(client,
                                             sku=Sku(name=sku),
                                             tags=tags,
                                             public_network_access=public_network_access,
-                                            disable_local_auth=disable_local_auth)
+                                            disable_local_auth=disable_local_auth,
+                                            soft_delete_retention_in_days=retention_days,
+                                            enable_purge_protection=enable_purge_protection)
+
+    return client.begin_create(resource_group_name, name, configstore_params)
+
 
+def recover_deleted_configstore(cmd, client, name, resource_group_name=None, location=None, yes=False):
+    if resource_group_name is None or location is None:
+        metadata_resource_group, metadata_location = resolve_deleted_store_metadata(cmd, name)
+
+        resource_group_name = resource_group_name if resource_group_name is not None else metadata_resource_group
+        location = location if location is not None else metadata_location
+
+    configstore_params = ConfigurationStore(location=location.lower(),
+                                            sku=Sku(name="Standard"),  # Only Standard SKU stores can be recovered!
+                                            create_mode=CreateMode.RECOVER)
```

Comment:
what is the original store is free? will it be recovered and upgraded? or server will fail the request

Reply:
You can't recover a free store because the retention period is 0 days.
So it won't be retained to be able to recover.

Reply:
If we hard code the standard sku here, if the store is actually a free one, what will be the error message from server? Basically CLI just need to error out that free store can't be recovered.